src/main/java/gut/follower/org/Controller/AccountController.java

```diff
@@ -0,0 +1,64 @@
+package gut.follower.org.Controller;
+
+import gut.follower.org.Models.Account;
+import gut.follower.org.Repositories.AccountRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/rest/account")
+public class AccountController {
+
+    @Autowired
+    AccountRepository repository;
+
+    @RequestMapping(method = RequestMethod.GET)
+    public Account getAccount(Principal principal){
+        return Optional
+                .ofNullable(repository.findByUsername(principal.getName()))
+                .orElseThrow(() -> new IllegalStateException("Account does not exist"));
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public Account updateAccount(Principal principal, @RequestBody Account account){
+        return Optional
+                .ofNullable(getAccountByName(principal))
+                .map(acc -> {
+                    acc.setPassword(getPassword(account, acc));
+                    acc.setEmail(getEmail(account, acc));
+                    return repository.save(acc);
+                })
+                .orElseThrow(() -> new IllegalStateException("Changes could not be applied"));
+    }
+
+    private String getEmail(Account account, Account acc) {
+        return Optional
```

Comment:
If you do as I showed above here you don't need to use Optional cos both values are already not null. Instead use casual if-else statement.